File: crawscript/film.go
package main

import "time"

type Film struct {
	Id                     int         `json:"id"`
	Title                  string      `json:"title"`
	StartDate              *time.Time  `json:"startDate"`
	EndDate                *time.Time  `json:"endDate"`
	Synopsis               string      `json:"synopsis"`
	Background             string      `json:"background"`
	Mean                   float64     `json:"mean"`
	Rank                   int         `json:"rank"`
	Popularity             int         `json:"popularity"`
	NumListUsers           int         `json:"numListUsers"`
	NumScoringUsers        int         `json:"numScoringUsers"`
	MediaType              int         `json:"mediaType"`
	State                  string      `json:"state"`
	MaxEpisodes            int         `json:"maxEpisodes"`
	NumEpisodes            int         `json:"numEpisodes"`
	Completed              bool        `json:"completed"`
	AverageEpisodeDuration int         `json:"averageEpisodeDuration"`
	Source                 string      `json:"source"`
	AgeRating              string      `json:"ageRating"`
	SerialId               int         `json:"serialId"`
	Studio                 Studio      `json:"studio"`
	Images                 []string    `json:"images"`
	Characters             []Character `json:"characters"`
}

//`id`                        int NOT NULL AUTO_INCREMENT,
//`title`                     varchar(255) NOT NULL,
//`start_date`                datetime DEFAULT CURRENT_TIMESTAMP,
//`end_date`                  datetime DEFAULT NULL,
//`synopsis`                  TEXT,
//`background`                TEXT,
//`mean`                      decimal(5,2) DEFAULT 0.00,
//`rank`                      int,
//`popularity`                int DEFAULT 0,
//`num_list_users`            int DEFAULT 0,
//`num_scoring_users`         int DEFAULT 0,
//`media_type`                varchar(50),
//`state`                     enum('upcoming','now_streaming', 'released', 'discontinued'),
//`max_episodes`              int,
//`num_episodes`              int,
//`completed`                 BOOLEAN NOT NULL DEFAULT FALSE,
//`average_episode_duration`  int,
//`source`                    varchar(50),
//`age_rating`                int,
//`images`                    JSON,
//`status`                    int DEFAULT 1,
//`created_at`                datetime DEFAULT CURRENT_TIMESTAMP,
//`updated_at`                datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
//
//`series_id`                 int,

File: api/film/src/main/java/site/ani4h/film/film/Film.java
package site.ani4h.film.film;

import site.ani4h.shared.common.Image;
import site.ani4h.shared.common.Uid;

import java.time.LocalDateTime;
import java.util.List;

public class Film {
    private Uid id;
    private String title;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private String synopsis;
    private List<Image> images;
    private Float avgStar;
    private int totalStar;
    private String state;                    //
    private int maxEpisodes;             // int (nullable)
    private int numEpisodes;             // int (nullable)
    private int year;                    // SMALLINT UNSIGNED (0-65535)
    private String season;                   // enum ('spring','summer','fall','winter') DEFAULT NULL
    private Float averageEpisodeDuration;    // float (nullable)
    private Uid ageRatingId;
    private int status;                  // int DEFAULT 1
    private LocalDateTime createdAt;         // datetime DEFAULT CURRENT_TIMESTAMP
    private LocalDateTime updatedAt;         // datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
}
